scipy/signal/_signaltools.py

```diff
@@ -2452,6 +2458,271 @@ def hilbert2(x, N=None):
     return x
 
 
+def envelope(z: np.ndarray, bp_in: tuple[int | None, int | None] = (1, None), *,
+             n_out: int | None = None, squared: bool = False,
+             residual: Literal['lowpass', 'all', None] = 'lowpass',
+             axis: int = -1) -> tuple[np.ndarray, np.ndarray] | np.ndarray:
+    r"""Compute the envelope of a real- or complex-valued signal after applying an
+    optional bandpass filter.
+
+    Any complex-valued signal `z(t)` can be split into a real-valued (instantaneous)
+    amplitude `a(t)` and a real-valued (instantaneous) phase `phi(t)`, i.e.,
+    `z(t) = a(t) * exp(1j*phi(t))`. The envelope is defined as the absolute value of
```

Comment:
The redundancy is intentional here.  In signal processing, when talking about phase and amplitude it is often implicitly assumed that those are  stationary. I think, the redundancy makes it easier to grasp for many people.

Reply:
Changed the wording slightly in commit f910d6e7

Reply:
I think it would be better to write phi in Greek letters. Could you write the math expressions in latex? If this is not possible, I would suggest denoting the phase with p instead of phi.

Reply:
I also like the LaTeX expression better, but not all IDEs render those correctly when displaying help texts. 

There is guidance on that topic in SciPy, though not very straightforward to find:
The [SciPy contributor Guide] links to the  [Numpy Documentation Style] , which links to [Numpydoc] guide
which states that LaTeX may be used in the [Notes section].

So I followed that convention.

[SciPy contributor Guide]: https://docs.scipy.org/doc/scipy/dev/contributor/contributor_toc.html#documentation
[Numpy Documentation Style]: https://numpy.org/devdocs/dev/howto-docs.html#howto-document
[Numpydoc]: https://numpydoc.readthedocs.io/en/latest/format.html
 [Notes section]: https://numpydoc.readthedocs.io/en/latest/format.html#notes

Reply:
I guess using `p(t)` instead of `phi(t)` is a matter of taste—I'm just used to signal processing literature where an phase angle is always denoted by a Greek letter.